### ISSUE TITLE
Add Support for Mongoose v8 in @casl/mongoose

### ISSUE THREAD

**Issue by mdwekat** [stalniy/casl#838]
**Is your feature request related to a problem? Please describe.**
Yes, the problem is related to compatibility issues. I'm using Mongoose v8, but the latest version of `@casl/mongoose` (7.2.1) only supports up to Mongoose v7. This causes compatibility issues for projects that have upgraded to the latest version of Mongoose.

**Describe the solution you'd like**
I would like `@casl/mongoose` to be updated to support Mongoose v8. Specifically, the peer dependency in the package should include mongoose v8.x.x, allowing seamless integration with projects using the latest Mongoose version.

**Describe alternatives you've considered** (optional)
To temporarily resolve the compatibility issue, I've used the `overrides` property in `package.json` to specifically set the `mongoose` dependency for `@casl/mongoose` to `^8.0.0`. This workaround allows me to use Mongoose v8 with `@casl/mongoose`, but it is not an ideal long-term solution. It could potentially lead to unforeseen issues since `@casl/mongoose` has not been officially tested or updated to work with Mongoose v8.

```json
"overrides": {
    "@casl/mongoose": {
      "mongoose": "^8.0.0"
    }
  }
```

**Comment by hansemannn** [comment]
Now that MongoDB will automatically update all basic clusters to MongoDB v7 next month and that one requires the node driver v6, being ready for Mongoose v8 would be very helpful to remain stable for many of the projects using this library.

**Comment by stalniy** [comment]
breaking changes do not affected casl, mostly, there is some new side-effect with conditions object. For some reason, mongoose or its deps modify array ref, that's why test failed. There was no such behavior previously

**Comment by stalniy** [comment]
available in @casl/mongoose@7.2.2
